### ISSUE TITLE
Support Vertical Areas

### ISSUE THREAD

**Issue by curran** [curran/d3-area-label#18]
Like this
![image](https://user-images.githubusercontent.com/68416/29201956-ddea4d84-7e80-11e7-94c9-921053b8d600.png)

from [Original StreamGraph Paper](https://www.cg.tuwien.ac.at/courses/InfoVis/papers/InfoVis2008%20Stacked%20Graphs%20-%20Geometry%20&%20Aesthetics%20[Byron].pdf)

**Comment by vchouksey** [comment]
wrapping of text feature is also not available

**Comment by curran** [comment]
Indeed, that would be a cool feature! Any ideas on how it could be implemented?

**Comment by vchouksey** [comment]
I tried to wrap using MBosstock wrapper function but not succeeded, here is the example where i used your d3 area label and MBosstock wrapper function https://jsfiddle.net/vishal_29/8zk9jqcg/

**Comment by curran** [comment]
Could you please clarify where this "MBosstock wrapper function" came from?

I think it can work. The label position function will need to know how many lines there are, and will need to know the bounding box of all the lines combined.

**Comment by curran** [comment]
Hey I got it working. It works if you apply the wrap function *before* positioning the labels. Here's a working jsFiddle https://jsfiddle.net/8zk9jqcg/1/

**Comment by vchouksey** [comment]
Wow, we added new feature...

**Comment by vchouksey** [comment]
![topic_river](https://user-images.githubusercontent.com/6507341/29663364-feb43c06-88ca-11e7-9c62-95f3a313f5a5.PNG)

Open this image, Does D3-AreaLabel can handle like labeling in image?

**Comment by curran** [comment]
Currently, it can position a single thing within an area, based on the bounding box of that thing. That single thing can contain multiple things, for example it could be an SVG `<g>` element with anything inside it.

From the image, I'm not sure what is really going on. But it seems there are multiple labels inside each stream, and those labels are each positioned individually. This is not something that D3-AreaLabel can do (yet).

**Comment by vchouksey** [comment]
https://jsfiddle.net/vkchouksey/axor08ph/8/

Here, words getting overlap with other layers

**Comment by curran** [comment]
Yeah, this happens when there's a low number of data points. It's because of the curvyness of `d3.curveBasis`. This is what it looks like if you switch it to `d3.curveLinear`:

![image](https://user-images.githubusercontent.com/68416/29714890-a965e080-89c2-11e7-8634-121b80239843.png)

The label placement algorithm does linear interpolation, so effectively it treats the curve as though it's using `d3.curveLinear`, even thought it may not be. I haven't figured out a way to use the actual curve - I would love to change it to use the real curve, but I'm not sure how.

**Comment by vchouksey** [comment]
Hi Curran,

Little stacked while labelling need little help.screenshot error in attachment.

<img width="1418" alt="screen shot 2017-08-31 at 10 55 52" src="https://user-images.githubusercontent.com/6507341/29915103-0d7fc730-8e3b-11e7-9f23-8540235b6227.png">

https://codepen.io/vchouksey/project/editor/XwKbMe

**Comment by vchouksey** [comment]
HI Curran,

Little stucked labelling is not working this time, I am attaching image of chrome debugging console screen and code on codepen.

https://codepen.io/vchouksey/project/editor/XwKbMe

<img width="1416" alt="screen shot 2017-08-31 at 11 03 23" src="https://user-images.githubusercontent.com/6507341/29915529-668ca022-8e3c-11e7-81a9-ef4515f54153.png">

**Comment by curran** [comment]
I got vertical areas working in sort of a hacky way here, with word wrapping https://bl.ocks.org/curran/c567efde3b3da7bc1f642627072aabfc

![image](https://user-images.githubusercontent.com/68416/31180741-069290a2-a93d-11e7-9831-cd3146d75f68.png)
